src/v/kafka/data/log_reader_config.h

```diff
@@ -0,0 +1,85 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+#pragma once
+
+#include "model/fundamental.h"
+#include "model/record_batch_types.h"
+#include "model/timestamp.h"
+#include "storage/offset_translator_state.h"
+#include "storage/types.h"
+
+#include <functional>
+#include <optional>
+
+namespace kafka {
+
+struct log_reader_config {
+    kafka::offset start_offset;
+    kafka::offset max_offset;
+    size_t min_bytes;
+    size_t max_bytes;
+
+    /// \brief gurantees first_timestamp >= record_batch.first_timestamp
+    /// it is the std::lower_bound
+    std::optional<model::timestamp> first_timestamp;
+
+    /// abort source for read operations
+    model::opt_abort_source_t abort_source;
+
+    // do not let the lower level readers go over budget even when that means
+    // that the reader will return no batches.
+    bool strict_max_bytes{false};
+
+    model::opt_client_address_t client_address;
+
+    log_reader_config(
+      kafka::offset start_offset,
+      kafka::offset max_offset,
+      size_t min_bytes,
+      size_t max_bytes,
+      std::optional<model::timestamp> time,
+      model::opt_abort_source_t as,
+      model::opt_client_address_t client_addr = std::nullopt)
+      : start_offset(start_offset)
+      , max_offset(max_offset)
+      , min_bytes(min_bytes)
+      , max_bytes(max_bytes)
+      , first_timestamp(time)
+      , abort_source(as)
+      , client_address(std::move(client_addr)) {}
+
+    /**
+     * Read offsets [start, end].
+     */
+    log_reader_config(
+      kafka::offset start_offset,
+      kafka::offset max_offset,
+      model::opt_abort_source_t as = std::nullopt,
+      model::opt_client_address_t client_addr = std::nullopt)
+      : log_reader_config(
+          start_offset,
+          max_offset,
+          0,
+          std::numeric_limits<size_t>::max(),
+          std::nullopt,
+          as,
+          std::move(client_addr)) {}
+
+    friend std::ostream& operator<<(std::ostream& o, const log_reader_config&);
+};
+
+// Empty, invalid reader config which is sometimes useful as a placeholder
+// since log_reader_config doesn't have a default constructor.
+inline const log_reader_config empty_reader_config{{}, {}};
+
+storage::local_log_reader_config kafka_to_local_log_reader_config(
+  log_reader_config, ss::lw_shared_ptr<const storage::offset_translator_state>);
```

Comment:
This seems like the only reason kafka/data:log_reader_config depends on storage -- consider flipping the dependency considering how lightweight the log_reader_config is otherwise (basically only depends on model). Maybe move this constructor into storage somewhere. Otherwise it seems like a needlessly heavy dependency.

Reply:
Alternatively, inline the calls. It's not a lot of code.